source/core_sntp_client.c

```diff
@@ -411,3 +410,78 @@ SntpStatus_t Sntp_SendTimeRequest( SntpContext_t * pContext,
 
     return status;
 }
+
+const char * Sntp_StatusToStr( SntpStatus_t status )
+{
+    const char * pString = NULL;
+
+    switch( status )
+    {
+        case SntpSuccess:
+            pString = "SntpSuccess";
+            break;
+
+        case SntpErrorBadParameter:
+            pString = "SntpErrorBadParameter";
+            break;
+
+        case SntpRejectedResponseChangeServer:
+            pString = "SntpRejectedResponseChangeServer";
+            break;
+
+        case SntpRejectedResponseRetryWithBackoff:
+            pString = "SntpRejectedResponseRetryWithBackoff";
+            break;
+
+        case SntpRejectedResponseOtherCode:
+            pString = "SntpRejectedResponseOtherCode";
+            break;
+
+        case SntpErrorBufferTooSmall:
+            pString = "SntpErrorBufferTooSmall";
+            break;
+
+        case SntpInvalidResponse:
+            pString = "SntpInvalidResponse";
+            break;
+
+        case SntpClockOffsetOverflow:
+            pString = "SntpClockOffsetOverflow";
+            break;
+
+        case SntpZeroPollInterval:
+            pString = "SntpZeroPollInterval";
+            break;
+
+        case SntpErrorTimeNotSupported:
+            pString = "SntpErrorTimeNotSupported";
+            break;
+
+        case SntpErrorChangeServer:
+            pString = "SntpErrorChangeServer";
+            break;
+
+        case SntpErrorDnsFailure:
+            pString = "SntpErrorDnsFailure";
+            break;
+
+        case SntpErrorNetworkFailure:
+            pString = "SntpErrorNetworkFailure";
+            break;
+
+        case SntpServerNotAuthenticated:
+            pString = "SntpServerNotAuthenticated";
+            break;
+
+        case SntpErrorAuthFailure:
+            pString = "SntpErrorAuthFailure";
+            break;
+
+        default:
+            pString = "Invalid status code!";
+            assert( false );
```

Comment:
Should there be an `assert` for an external API? This assert will also trigger as part of the unit tests if asserts aren't disabled.